src/command/expire/expire.c

```diff
@@ -59,17 +59,21 @@ expireBackup(InfoBackup *infoBackup, String *removeBackupLabel, String *backupEx
     ASSERT(removeBackupLabel != NULL);
     ASSERT(backupExpired != NULL);
 
-    storageRemoveNP(storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s/" BACKUP_MANIFEST_FILE, strPtr(removeBackupLabel)));
-    storageRemoveNP(
-        storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s/" BACKUP_MANIFEST_FILE INFO_COPY_EXT, strPtr(removeBackupLabel)));
+    // execute the real expiration and deletion
+    // only if the dry-run mode is disabled
+    if ( ! cfgOptionBool(cfgOptDryRun) ){
+      storageRemoveNP(storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s/" BACKUP_MANIFEST_FILE, strPtr(removeBackupLabel)));
+      storageRemoveNP(
+                      storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s/" BACKUP_MANIFEST_FILE INFO_COPY_EXT, strPtr(removeBackupLabel)));
 
-    // Remove the backup from the info file
-    infoBackupDataDelete(infoBackup, removeBackupLabel);
+      // Remove the backup from the info file
+      infoBackupDataDelete(infoBackup, removeBackupLabel);
```

Comment:
Ah, I see that I made a bad comment here (sorry!): infoBackupDataDelete only removes the backup from the in-memory copy of the backup.info file not the actual file. So here we do want to always run infoBackupDataDelete whether we are doing a dry-run or not. The in-memory infoBackup will then be updated and that updated information will be used by the removeExpiredArchive and removeExpiredBackup commands.

Reply:
got it, thanks